astroquery/hips2fits/core.py

```diff
@@ -0,0 +1,408 @@
+        response = self.query_without_wcs_async(get_query_payload, hips=hips, width=width, height=height, projection=projection, ra=ra, dec=dec, fov=fov, coordsys=coordsys, rotation_angle=rotation_angle, format=format, min_cut=min_cut, max_cut=max_cut, stretch=stretch, cmap=cmap)
+
+        if get_query_payload:
+            return response
+
+        result = self._parse_result(response, verbose=verbose, format=format)
+        return result
+
+
+    @class_or_instance
+    def query_without_wcs_async(self, get_query_payload=False, **kwargs):
+        request_payload = self._args_to_payload(**kwargs)
+        
+        # primarily for debug purposes, but also useful if you want to send
+        # someone a URL linking directly to the data
+        if get_query_payload:
+            return request_payload
+
+        response = self._request(
+            method="GET",
+            url=self.server,
+            data=kwargs.get('data', None),
+            cache=False,
+            timeout=self.timeout,
+            params=request_payload
+        )
+
+        return response
+
+    def _parse_result(self, response, verbose, format):
+        # Parse the json string result
+        if response.status_code != 200:
+            # Error code status
+            content = response.json()
+            # Print a good user message based on what the
+            # server returned
+            # The server returns a dict with two items:
+            # - The title of the error
+            # - A little description of the error
+            msg = content['title']
+            if 'description' in content.keys():
+                msg += ': ' + content['description'] + '\n'
+
+            raise AttributeError(msg)
+        else:
+            # The request succeeded
+            bytes_str = response.content
+            # Check the format. At this point, we are sure
+            # that the format is correct i.e. either, fits, jpg or png
+            if format == 'fits':
+                hdul = fits.HDUList.fromstring(bytes_str)
+                return hdul
+            else:
+                # jpg/png formats
+                bytes = io.BytesIO(bytes_str)
+                im = Image.open(bytes)
+                try:
+                    data = np.asarray(im)
+                    return data
+                except:
+                    ValueError('The image data cannot be extracted from the image itself \n')
+
+    def _args_to_payload(self, **kwargs):
+        # Convert arguments to a valid requests payload
+        # Build the payload
+        payload = {}
+
+        # If a wcs is still present in the kwargs it means
+        # the user called query_with_wcs
+        if kwargs.get('wcs'):
+            wcs = kwargs.pop('wcs')
+            header = wcs.to_header()
+
+            # The WCS header does not contain the NAXISx keywords
+            # We can get them from the WCS directly and add them
+            # to the header
+            nx, ny = wcs.pixel_shape
```

Comment:
you may not be able to rely on this being present; if a wcs was created not from a fits Header (i.e., from scratch), it won't have this attribute populated

Reply:
The hips2fits service asks for the size of the result image (width and height), it is mandatory. I changed the code, if the wcs has no pixel shape (pixel_shape  None) then I raise an AttributeError saying either to give your wcs the pixel size of the image or to use the other **query** method

Reply:
50f14de